README.md

```diff
@@ -1,19 +1,111 @@
+
+**Versioning:** We saw this with the Foursquare API. Versioning is an
+   important consideration, and you have to make sure that you don't
+release code in such a way that it suddenly breaks applications that
+rely on your API.
+
+**Authentication and Security:** Allowing external access to your data is a
+   serious thing, so your external API needs robust access control and
+security.
+
+**Quality of Service:** Once you have third-party developers relying
+   on your API, there are expectations around quality, uptime, response
+time.
+
+You also have to consider things like **rate limiting**, where you
+monitor access and only allow a certain number of API calls per
+application in a given timeframe.
+
+We won't be covering much of this in this unit, but these are all
+important considerations if you plan to make your API public.
+
+### Internal API Considerations
+
+An internal API is a great way to provide access to data for multiple internal
+applications. While there are still many things to consider, the
+concerns for internal APIs are generally less strenuous because you are
+usually in control of all of the applications that will have access.
+
+**Documentation:** Documentation is still important, but with an
+   internal API you generally won't have to provide support for such a
+wide array of consumers, and the docs can be a little more utilitarian.
+
+**Versioning:** Versioning is still a consideration. Depending on
+   your development process, it's possible that not all of your apps
+will be able to respond to changes at the same time, so you may still
+need versioning to support that. However, the consumers will all be
+known, so you won't have to worry as much about some rogue developer
+steadfastly holding on to the beta version of your API.
+
+**Authentication and Security:** Security is still a concern, of
+   course, but again, you'll be in control of what applications can
+access your API.
+
+We don't want to store access keys or passwords on the client of your
+internal API in cookies or JavaScript, because then it wouldn't be secure. And depending on the
+use of the API, it might not make sense to do user-based authentication
+at all. So how do we ensure that only valid web applications are
+accessing our internal API?
+
+Browsers implement something called [Same Origin Security Policy](https://en.wikipedia.org/wiki/Same-origin_policy), which prevents a page from running a script to access data from a different origin.
+
+This is great because by default it means your API can't be accessed by
+another domain. But what about the ones you want to give access to? For
+that, we can configure [Cross Origin Resource Sharing](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing), or **CORS**, to allow whitelist access to our domains while still blocking others. We won't be doing much with this in this unit, but there are great tools like [rack-cors](https://github.com/cyu/rack-cors) out there to help.
```

Comment:
Trying to put myself in the shoes of a beginning user, I'm a bit confused at the end of this section. How do Same Origin Security Policy and CORS work together to add security? Can we simplify how we describe these tools for users?

For example, even the term "whitelist access" feels a bit jargon-y here. Can we reword this description to make it easier to understand?